src/test/java/com/puppycrawl/tools/checkstyle/DetailNodeTreeStringPrinterTest.java

```diff
@@ -37,10 +37,17 @@
 import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 
-public class DetailNodeTreeStringPrinterTest {
+public class DetailNodeTreeStringPrinterTest extends AbstractTreeTestSupport {
 
-    private static String getPath(String filename) {
-        return "src/test/resources/com/puppycrawl/tools/checkstyle/astprinter/" + filename;
+    // [REFLECTION]
+    // DetailNodeTreeStringPrinter#getParseErrorMessage is used for creating error messages
+    // for validating those obtained in UTs against the ones created.
+    private static final Method GET_PARSE_ERROR_MESSAGE = Whitebox.getMethod(
+            DetailNodeTreeStringPrinter.class, "getParseErrorMessage", ParseErrorMessage.class);
```

Comment:
Why it is not possible to make bad javadoc files and give them to DetailNodeTreeStringPrinter and recheck output of it ?
Please avoid reflection as much as possible. Please write more functional tests.